assets/app/data/services/upgrade/openstack-services.factory.js

```diff
@@ -0,0 +1,26 @@
+(function() {
+
+    angular
+        .module('crowbarData.upgrade')
+        .factory('upgradeOpenStackServicesFactory', upgradeOpenStackServicesFactory);
+
+    upgradeOpenStackServicesFactory.$inject = ['$q', '$http'];
+    /* @ngInject */
+    function upgradeOpenStackServicesFactory($q, $http) {
+        var factory = {
+            getOpenStackServices: getOpenStackServices
```

Comment:
Here we could centralise the interaction with OpenStack, not just for the upgrade methods... So I'd change the name to `openStackFactory`.
Also, on this page we need to perform two actions: Stop Openstack Services, and create Openstack Backup.
We could create two methods in this factory to achieve it.